velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -28,13 +28,29 @@
 #include <aws/core/auth/AWSCredentialsProviderChain.h>
 #include <aws/core/http/HttpResponse.h>
 #include <aws/core/utils/logging/ConsoleLogSystem.h>
+#include <aws/core/utils/stream/PreallocatedStreamBuf.h>
 #include <aws/identity-management/auth/STSAssumeRoleCredentialsProvider.h>
 #include <aws/s3/S3Client.h>
 #include <aws/s3/model/GetObjectRequest.h>
 #include <aws/s3/model/HeadObjectRequest.h>
 
 namespace facebook::velox {
 namespace {
+// Reference: https://issues.apache.org/jira/browse/ARROW-8692
```

Comment:
I looked at the link, but couldn't find any text that would clarify what this is about. Would you summarize?

Reply:
The reference is in the PR link. Specifically at the line below. Should I put the below link then?
https://github.com/apache/arrow/pull/7098/files#diff-27b767cd06b879c40106ea98d21d3b12f13ec6e9f9aa49218d10583c5c1899c4R363

Reply:
This link is more information. Perhaps, summarize the necessary information here and optionally add a link. A link by itself is not very informative and requires a reader to just to the link and then come back.

Reply:
I summarized the exact block from Arrow here and included the reference as well.